### PR TITLE
TGMC Christmas 2023 [IDB IGNORE]

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

a_xenomorph_queen_wearing_a_santa_suit_a

With Thanksgiving officially over, it's time to move onto the next holiday.
Introducing the TGMC Christmas PR, this PR

-Adds festive decorations to all ships and shuttles.
-Adds Christmas food and drink to all ships and shuttles.
-Places Christmas trees throughout the ships.
-Implements gifts with randomly generated presents inside, much like TG already does (for the sake of balance the items are much more curated compared to TG.)
-Allows players the magical chance to meet Santa and his elves face to face.
-Various other surprises and things I can't be bothered to note.

This PR isn't meant to be merged, only TMC'ed. At a later date I _might_ add some of the holiday stuff into the code itself for easier reuse in future Christmases.

**TO ADMINS: Add this TM into the game whenever you think is appropriate for Christmas time, this PR was opened somewhat early so the code could be vetted before December rolls around**

## Why It's Good For The Game

We didn't have Christmas in 2021, I have taken it upon myself to fix this problem. If you don't like this PR you are officially a scrooge who needs to embark on a vaguely allegorical journey to learn the true meaning of Christmas.

## Changelog
:cl:
add: Added Christmas. If you're seeing this in game the PR was merged somehow, whoops.
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
do we also get xeno chrismtas skins like last yera

**Comment by RipGrayson** [comment]
> do we also get xeno chrismtas skins like last yera

Different PR, convince Splinter to reopen 
https://github.com/tgstation/TerraGov-Marine-Corps/pull/5568 this year, shouldn't be any conflicts

**Comment by FosterGrime-Heymynameslouie** [comment]
> > do we also get xeno chrismtas skins like last yera
> 
> Different PR, convince Splinter to reopen #5568 this year, shouldn't be any conflicts

its that time of the year again

**Comment by iforgotmeword** [comment]
looks good i think you guys don't get this a lot but really good job trying to make this game just little butter and funnier for most(98%) of people

File: code/game/objects/items/gift.dm [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+				GLOB.round_statistics.presents_grinched += 1
+				if(!HAS_TRAIT(M, TRAIT_CHRISTMAS_GRINCH))
+					GLOB.round_statistics.number_of_grinches += 1
+					ADD_TRAIT(M, TRAIT_CHRISTMAS_GRINCH, TRAIT_CHRISTMAS_GRINCH) //bad present openers are effectively cursed to receive nothing but coal for the rest of the round
+					to_chat(M, span_boldannounce("Your heart feels three sizes smaller..."))
+				spawnpresent(M) //they have the grinch trait, the presents will always spawn coal
+			else
+				spawnpresent(M, TRUE) //they got lucky, the present will open as normal but with a STOLEN label in the desc
+			qdel(src)
+		return
```
Comment (psykzz) [inline code/game/objects/items/gift.dm:62]: inverting this check would do a lot to making it a lot more readable, if you care about (considering this is a xmas PR)

**Comment by zzurge** [comment]
The most wonderful time of the year...

**Comment by FosterGrime-Heymynameslouie** [comment]
its that time of the year again

**Comment by RipGrayson** [comment]
Well this was fun, maybe next year

**Comment by novaepee** [comment]
SANTA CLAUSE WAS REAAAAAAALLLLLLLLLLLLL

**Comment by RipGrayson** [comment]
Reopening this for code review, anticipate TM somewhere in early-ish December

**Comment by Kuro020** [comment]
Imagine playing tgmc during christmas

**Comment by Technical-l** [comment]
festive35years (1)

**Comment by Helg2** [comment]
I think the sentinel sprite is a bit buggy, because there's no santa hat sprited on the walking animations.

**Comment by Helg2** [comment]
Also i think you lost an icon file for fruit cakes.

**Comment by RipGrayson** [comment]
> Also I think you lost an icon file for fruit cakes, pumpkin cakes and apple cakes.
Technically I didn't lose the file so much as the icons were migrated to other parts in a different PR and I forgot to fix it for this PR until you mentioned it. Thanks for the report!

**Comment by RipGrayson** [comment]
> I think the sentinel sprite is a bit buggy, because there's no santa hat sprited on the walking animations.

Also fixed

**Comment by Helg2** [comment]
Also, can you change `/obj/item/reagent_containers/food/snacks/mre_pack/xmas3` name's from `"\improper Xmas Prepared Meal:fruitcake"` to `"\improper Xmas Prepared Meal (fruitcake)"` so it keeps the general format? And so on with other xmas meals

**Comment by RipGrayson** [comment]
Christmas is over, time for this PR to hibernate for another year.

**Pull request closed without merging**